### PR TITLE
chore: optimize the apisix cluster processing

### PR DESCRIPTION
**Pull request by tokers**
The processing for the apisix cluster was optimized, so it can be easier to implement the multiple clusters support in the future.

### CODE REVIEW

File: pkg/ingress/controller.go [hunk 1]
```diff
@@ -318,17 +318,17 @@ func (c *Controller) run(ctx context.Context) {
 	c.metricsCollector.ResetLeader(true)
 
 	err := c.apisix.AddCluster(&apisix.ClusterOptions{
-		Name:     "",
-		AdminKey: c.cfg.APISIX.AdminKey,
-		BaseURL:  c.cfg.APISIX.BaseURL,
+		Name:     c.cfg.APISIX.DefaultClusterName,
+		AdminKey: c.cfg.APISIX.DefaultClusterAdminKey,
+		BaseURL:  c.cfg.APISIX.DefaultClusterBaseURL,
```
Comment (tao12345666333) [inline pkg/ingress/controller.go:323]: Is this a breaking change?

File: pkg/config/config.go [hunk 2]
```diff
@@ -78,8 +78,18 @@ type KubernetesConfig struct {
...
+	// DefaultClusterName is the name of default cluster.
+	DefaultClusterName string `json:"default_cluster_name"`
+	// DefaultClusterBaseURL is the base url configuration for the default cluster.
+	DefaultClusterBaseURL string `json:"default_cluster_base_url" yaml:"default_cluster_base_url"`
+	// DefaultClusterAdminKey is the admin key for the default cluster.
 	// TODO: Obsolete the plain way to specify admin_key, which is insecure.
+	DefaultClusterAdminKey string `json:"default_cluster_admin_key" yaml:"default_cluster_admin_key"`
+	// BaseURL is same to DefaultClusterBaseURL.
+	// Deprecated: use DefaultClusterBaseURL instead.
+	BaseURL string `json:"base_url" yaml:"base_url"`
```
Comment (tao12345666333) [inline pkg/config/config.go:90]: Should we have a deprecation policy?  e.g. when to actually remove it.

File: pkg/ingress/controller.go (hunk 1 above)
Comment (tokers) [reply pkg/ingress/controller.go:323]: Not really, we still reserve `BaseURL` and `AdminKey`.

File: pkg/config/config.go (hunk 2 above)
Comment (tokers) [reply pkg/config/config.go:90]: What about:

```
// Deprecated: use DefaultClusterBaseURL instead, BaseURL will be removed once v1.0.0 is released.
```

Comment (tao12345666333) [reply pkg/config/config.go:90]: SGTM.

Comment (tokers) [reply pkg/config/config.go:91]: Changed.

Comment (tokers) [reply pkg/config/config.go:91]: Changed.

**Comment by gxthrj** [comment]
I have two questions.
How to configure multi-clusters ?  
How to match cluster `name` and `admin url` and `key`.

**Comment by tokers** [comment]
> I have two questions.
> How to configure multi-clusters ?
> How to match cluster `name` and `admin url` and `key`?

Not in this PR.

**Comment by gxthrj** [comment]
> > I have two questions.
> > How to configure multi-clusters ?
> > How to match cluster `name` and `admin url` and `key`?
> 
> Not in this PR. See the issue description, multiple-clusters will be supported in the future, this PR is just one of the prerequisites.

It is difficult to set up a multi-cluster according to the current configuration. 

The user modified the configuration according to `defaultxxx` this time, and the next time the multi-cluster is actually configured, it will be modified again. 

It is hoped that this modification can finalize the configuration file structure.

**Comment by gxthrj** [comment]
Got it, with `ApisixClusterConfig` CRD .

**Review by gxthrj — APPROVED** [review]

**Pull request merged**